lightning/src/ln/outbound_payment.rs

```diff
@@ -947,6 +947,8 @@ impl OutboundPayments {
 					let amount_msat = match InvoiceBuilder::<DerivedSigningPubkey>::amount_msats(&invreq) {
 						Ok(amt) => amt,
 						Err(_) => {
+							// We check this during invoice request parsing, when constructing the invreq's
+							// contents from its TLV stream.
 							debug_assert!(false, "LDK requires an invreq amount for amount-less offers");
```

Comment:
More so the spec requires it.

Reply:
It looks like invreqs may omit their amount if the offer has an amount, so if that offer amount were `Currency` in theory we could be spec compliant but still hit this case. Let me know if I'm missing something.

Reply:
Oh, I see what you're saying actually. Will rephrase this.